peerprep-backend/src/main/java/com/peerprep/peerprepbackend/controller/MatchController.java

```diff
@@ -0,0 +1,32 @@
+package com.peerprep.peerprepbackend.controller;
+
+import com.peerprep.peerprepbackend.dto.request.TestWebsocketMessageRequest;
+import com.peerprep.peerprepbackend.dto.response.TestWebsocketReplyResponse;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.messaging.handler.annotation.MessageMapping;
+import org.springframework.messaging.simp.SimpMessagingTemplate;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.util.HtmlUtils;
+
+
+@RestController
+@RequiredArgsConstructor
+public class MatchController {
+
+    @Autowired
+    private SimpMessagingTemplate simpMessagingTemplate;
+
+    @MessageMapping("/match")
+    public void greeting(TestWebsocketMessageRequest message) throws Exception {
+        Thread.sleep(1000);
+        this.simpMessagingTemplate.convertAndSend("/topic/matching", new TestWebsocketReplyResponse("from backend websocket: " + HtmlUtils.htmlEscape(message.getMessage())));
```

Comment:
can use `@SendTo` annotation to avoid using SimpMessagingTemplate directly, just tried this and it works

```
    @MessageMapping("/match")
    @SendTo("/topic/matching")
    public TestWebsocketReplyResponse greeting(TestWebsocketMessageRequest message) throws Exception {
        Thread.sleep(1000);
        return new TestWebsocketReplyResponse("from backend websocket: " + HtmlUtils.htmlEscape(message.getMessage()));
    }

Reply:
yeah I think this was the initial implementation but I changed it to  SimpMessagingTemplate for some reason that I cannot recall. Ima change it back and we'll cross the bridge when we get there